realm/realm-library/src/androidTest/java/io/realm/RealmJsonNullPrimaryKeyTests.java

```diff
@@ -130,23 +125,28 @@ public void createOrUpdateObjectFromJson_primaryKey_isNullOrAbsent_fromJsonObjec
         }
     }
 
-    // Testing null or absent primary key value for createObject() -> createOrUpdateObjectFromJson()
+    // Testing null primary key value for createObject() -> createOrUpdateObjectFromJson()
     @Test
-    public void createOrUpdateObjectFromJson_primaryKey_isNullOrAbsent_updateFromJsonObject() throws JSONException {
-        realm.beginTransaction();
-        realm.createObject(clazz); // name = null, id = 0
-        realm.createOrUpdateObjectFromJson(clazz, new JSONObject(jsonString));
-        realm.commitTransaction();
-
+    public void createOrUpdateObjectFromJson_primaryKey_isNull_updateFromJsonObject() throws JSONException {
         // PrimaryKeyAsString
         if (clazz.equals(PrimaryKeyAsString.class)) {
+            realm.beginTransaction();
+            realm.createObject(clazz, null); // name = null, id =null
+            realm.createOrUpdateObjectFromJson(clazz, new JSONObject(jsonString));
+            realm.commitTransaction();
+
```

Comment:
HA, right, will change it. It was caused by that i thought the default value is `0` and `""`.